File: WebAppClients/src/types/customer.ts
export type DocumentType = 'DNI' | 'Pasaporte' | 'CE'

export interface Customer {
  id: number
  nombres: string
  apellidos: string
  documentType: DocumentType
  numeroDocumento: string
  fechaNacimiento: string
  nationality: string | null
  email: string
  telefono: string
  direccion: string
}

export type CustomerInput = Omit<Customer, 'id'>

File: WebAppClients.Api/Models/Customer.cs
using System.ComponentModel.DataAnnotations;

namespace WebAppClients.Api.Models;

public class Customer
{
    public int Id { get; set; }

    [Required, MaxLength(100)]
    public string Nombres { get; set; } = string.Empty;

    [Required, MaxLength(100)]
    public string Apellidos { get; set; } = string.Empty;

    [Required]
    public DocumentType DocumentType { get; set; }

    [Required, MaxLength(20)]
    public string NumeroDocumento { get; set; } = string.Empty;

    [Required]
    public DateOnly FechaNacimiento { get; set; }

    [MaxLength(100)]
    public string? Nationality { get; set; }

    [Required, EmailAddress, MaxLength(150)]
    public string Email { get; set; } = string.Empty;

    [Required, MaxLength(20)]
    public string Telefono { get; set; } = string.Empty;

    [Required, MaxLength(200)]
    public string Direccion { get; set; } = string.Empty;
}
